### PR TITLE
Fix pain points with Meetup event import

### PR DESCRIPTION
**Pull request by GuruKhalsa**
This addresses #251.  I included the group name in the title using the format "group name - event title", but please let me know if it should be formatted differently.
An end time is now being assigned to Events if duration is given.  Meetup recommends estimating a duration of 3 hours if none is given but discussion around issue #16 seemed to indicate that leaving it blank was preferable so that's how I implemented it.
Meetup group tags are now being imported as well and added to the Event's tag list.
My impression is that the Parsers are setup to import multiple Events if they are found, which is probably why the import redirects to the main page instead of an imported Event, but I updated it so that if only one Event is imported it will redirect to that newly imported Event's page.

### CODE REVIEW

**Comment by reidab** [comment]
Thanks! This is looking great. Can you add yourself to our `CONTRIBUTORS.md`?

**Comment by GuruKhalsa** [comment]
Thanks!  I updated CONTRIBUTERS.md.  While working on this issue I noticed that the Meetup API endpoints being used are deprecated, would it be useful for me to work on updating that and would it fall under the scope of this issue?

**Comment by reidab** [comment]
If these old endpoints are still working, I think it makes the most sense to accept this PR as-is and then open a new issue to track the updates. If you're interested in jumping in on that, please do!

File: app/controllers/calagator/sources_controller.rb [hunk 1]
```diff
@@ -7,7 +7,11 @@ def import
     @importer = Source::Importer.new(params.permit![:source])
     respond_to do |format|
       if @importer.import
-        format.html { redirect_to events_path, flash: { success: render_to_string(layout: false) } }
+        if @importer.events.count == 1
+          format.html { redirect_to @importer.events.first, flash: { success: render_to_string(layout: false) } }
+        else
+          format.html { redirect_to events_path, flash: { success: render_to_string(layout: false) } }
```
Comment (botandrose) [inline app/controllers/calagator/sources_controller.rb]: Quite a bit of duplication between these two lines. I suggest extracting a local variable for the redirect target to DRY this up.

File: app/models/calagator/source/parser/meetup.rb [hunk 2]
```diff
@@ -7,15 +7,19 @@ class Source::Parser::Meetup < Source::Parser
   def to_events
     return fallback unless Calagator.meetup_api_key.present?
     return unless data = get_data
+    start_time = Time.at(data['time']/1000).utc
+    group_topics = data['group']['topics']
+    topics_string = group_topics.map{ |t| t['name'].downcase }.join(', ').insert(0, ', ') unless group_topics.empty?
```
Comment (botandrose) [inline app/models/calagator/source/parser/meetup.rb]: This method is getting pretty complex, especially this line. I'd consider extracting some private methods, perhaps.

**Comment by GuruKhalsa** [comment]
Sounds good, the old endpoints are still working so I'll open a new issue.  One problem mentioned in #251 that I was unable to reproduce was 'Calagator rejecting edit changes because of too many links in the description (spam filter)'.  I wasn't able to find an event that created this issue so I didn't make any changes to address it.

**Comment by reidab** [comment]
Ah, I'd forgotten about that issue and can provide an example. It's triggered anytime you import a meetup event that has more than three URLs in the description, such as http://www.meetup.com/laruby/events/226450232/. The event will import just fine, but subsequent attempts to edit the event will fail because the description contains too many URLs.

That said, I'm not sure what the ideal solution to this would be. :confused:

File: app/controllers/calagator/sources_controller.rb (hunk 1 above)
Comment (reidab) [inline app/controllers/calagator/sources_controller.rb]: Agreed :+1:

Comment (GuruKhalsa) [inline app/controllers/calagator/sources_controller.rb]: Good call, I've updated it.  Thanks for the feedback!

**Comment by GuruKhalsa** [comment]
Without altering the spam filter guidelines my only thought would be to parse the description with RegEx and convert some of the links into plain text.

**Comment by reidab** [comment]
Since the spam filter detects URLs, not explicit `<a>` tags, this would be tricky. I'll pull this topic out of #251 into a separate issue, since it affects all import types.

File: app/controllers/calagator/sources_controller.rb [hunk 3]
```diff
@@ -7,7 +7,8 @@ def import
     @importer = Source::Importer.new(params.permit![:source])
     respond_to do |format|
       if @importer.import
-        format.html { redirect_to events_path, flash: { success: render_to_string(layout: false) } }
+        redirect_target = @importer.events.count == 1 ? @importer.events.first : events_path
```
Comment (botandrose) [inline app/controllers/calagator/sources_controller.rb]: Neat Ruby trick: `Enumerable` has a `#one?` method, so you could shorten the check to to `@importer.events.one?`. Not a showstopper by any means, just neat.

Comment (GuruKhalsa) [inline app/controllers/calagator/sources_controller.rb]: That is neat and more readable. Thanks!

**Comment by botandrose** [comment]
LGTM! :shipit:

**Pull request merged**